src/pipeline_edsl/to_workflow.ml

```diff
@@ -255,41 +253,116 @@ module Defaults = struct
   let results_dir = None
 end
 
+module Provenance_description = struct
+
+  type t = {
+    name: string;
+    sub_tree_arguments: (string * t) list;
```

Comment:
I see how we will make use of the `string` and `json` arguments, but what is the `sub_tree` for?